services/web/server/src/simcore_service_webserver/computation_comp_tasks_listening_task.py

```diff
@@ -78,28 +80,53 @@ async def listen(app: web.Application):
+            # and comp_tasks/comp_pipeline get deprecated.
+
             # find the user(s) linked to that project
-            joint_table = user_to_projects.join(projects)
-            query = (
-                select([user_to_projects])
-                .select_from(joint_table)
-                .where(projects.c.uuid == project_id)
+            result = await conn.execute(
+                select([projects]).where(projects.c.uuid == project_id)
             )
-            async for row in conn.execute(query):
-                user_id = row["user_id"]
-                try:
-                    node_data = await projects_api.update_project_node_outputs(
-                        app, user_id, project_id, node_id, data=task_output
-                    )
-                except projects_exceptions.ProjectNotFoundError:
-                    log.exception("Project %s not found", project_id)
-                except projects_exceptions.NodeNotFoundError:
-                    log.exception(
-                        "Node %s ib project %s not found", node_id, project_id
-                    )
-
-                messages = {"nodeUpdated": {"Node": node_id, "Data": node_data}}
-                await post_messages(app, user_id, messages)
+            the_project: RowProxy = result.fetchone()
+            if not the_project:
+                log.warning(
+                    "Project %s was not found and cannot be updated", project_id
+                )
+                continue
+            the_project_owner = the_project["prj_owner"]
+
+            # update the project
+            try:
+                node_data = await projects_api.update_project_node_outputs(
+                    app, the_project_owner, project_id, node_id, data=task_output
+                )
+            except projects_exceptions.ProjectNotFoundError:
+                log.exception(
+                    "Project %s was not found and cannot be updated", project_id
+                )
+                continue
+            except projects_exceptions.NodeNotFoundError:
+                log.exception(
+                    "Node %s ib project %s not found and cannot be updated",
+                    node_id,
+                    project_id,
+                )
+                continue
+            # notify the client(s), the owner + any one with read writes
+            clients = [the_project_owner]
+            for gid, access_rights in the_project["access_rights"].items():
+                if not access_rights["read"]:
+                    continue
+                # let's get the users in that group
+                async for user in conn.execute(
+                    select([user_to_groups.c.uid]).where(user_to_groups.c.gid == gid)
+                ):
+                    clients.append(user["uid"])
+
+            messages = {"nodeUpdated": {"Node": node_id, "Data": node_data}}
+            for client in clients:
```

Comment:
``asyncio.gather`` would go better here? You cannot anticipate how saturated the socket communication  with a client is ... and this would block notifications of others...

Reply:
well I can actuallly it is a small optimitzation, but post as the name imply is non blocking it is a fire and forget thing. it should not be an issue. will fix.